pydantic_ai_graph/pydantic_ai_graph/state.py

```diff
@@ -62,10 +84,17 @@ class EndEvent(Generic[StateT, RunEndT]):
 
     def __post_init__(self):
         # Copy the state to prevent it from being modified by other code
-        self.state = copy.deepcopy(self.state)
+        self.state = _deep_copy_state(self.state)
 
     def node_summary(self) -> str:
         return str(self.result)
 
 
-StepOrEnd = Union[Step[StateT, RunEndT], EndEvent[StateT, RunEndT]]
+def _deep_copy_state(state: StateT) -> StateT:
+    if state is None:
+        return None  # pyright: ignore[reportReturnType]
```

Comment:
changing this to `return state` might eliminate the type error without changing the logic